src/Mayaqua/DNS.c

```diff
@@ -405,8 +405,16 @@ void DnsResolver(THREAD *t, void *param)
 	struct addrinfo hints;
 	Zero(&hints, sizeof(hints));
 
-	hints.ai_family = AF_INET6;
-	hints.ai_flags = AI_ALL | AI_ADDRCONFIG | AI_V4MAPPED;
+	if (HasIPv6Address())
+	{
+		hints.ai_family = AF_INET6;
+		hints.ai_flags = AI_ALL | AI_ADDRCONFIG | AI_V4MAPPED;
+	}
+	else
+	{
+		hints.ai_family = AF_INET;
+		hints.ai_flags = AI_ADDRCONFIG;
```

Comment:
```suggestion
```

Reply:
Doing this in IPv4-only environment is unnecessary as AF_INET can only return IPv4.